Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -338,10 +347,51 @@ namespace PhysX
     }
 
     // DisplayCallback
-    void EditorHeightfieldColliderComponent::Display(AzFramework::DebugDisplayRequests& debugDisplay) const
+    void EditorHeightfieldColliderComponent::Display(const AzFramework::ViewportInfo& viewportInfo,
+        AzFramework::DebugDisplayRequests& debugDisplay) const
     {
-        const auto& heightfieldConfig = static_cast<const Physics::HeightfieldShapeConfiguration&>(*m_shapeConfig);
-        m_colliderDebugDraw.DrawHeightfield(debugDisplay, m_colliderConfig, heightfieldConfig);
+        AzPhysics::SimulatedBody* simulatedBody = GetSimulatedBody();
```

Comment:
`simulatedBody` can be const, which is great because that means the const version of `GetSimulatedBody` can return a const pointer too.

The cast to `staticRigidBody` can be const too.

Reply:
Not really, we call `StaticRigidBody::GetShape()` which is non-const. 

Interestingly even `StaticRigidBody::GetShapeCount()` is a non-const function. 
Hard to be const-correct when the interfaces aren't, or have to do const_casts everywhere

Reply:
Pity when this happens. If it's easy to add the const version of these functions please do, but if it's not trivial because they come from ebuses or something, just let it be.

Reply:
+1, would be great to update the functions to `const` or add `const` overloads